File: pjecalc-fonte/negocio/br/jus/trt8/pjecalc/negocio/dominio/calculo/inss/sobresalarios/OcorrenciaDeInssSalarioDevidoOptimizerListSearchUnique.java
/*
 * Decompiled with CFR 0.152.
 */
package br.jus.trt8.pjecalc.negocio.dominio.calculo.inss.sobresalarios;

import br.jus.trt8.pjecalc.negocio.comum.OptimizerListSearchUnique;
import br.jus.trt8.pjecalc.negocio.dominio.calculo.inss.sobresalarios.OcorrenciaDeInssSobreSalariosDevidos;
import br.jus.trt8.pjecalc.negocio.dominio.calculo.inss.sobresalarios.OcorrenciaInssUnique;
import java.util.Collection;
import java.util.HashMap;
import java.util.Map;

public class OcorrenciaDeInssSalarioDevidoOptimizerListSearchUnique
extends OptimizerListSearchUnique<OcorrenciaInssUnique, OcorrenciaDeInssSobreSalariosDevidos> {
    private Map<OcorrenciaInssUnique, OcorrenciaDeInssSobreSalariosDevidos> map;

    @Override
    public OptimizerListSearchUnique<OcorrenciaInssUnique, OcorrenciaDeInssSobreSalariosDevidos> init(Collection<OcorrenciaDeInssSobreSalariosDevidos> collection) {
        this.map = new HashMap<OcorrenciaInssUnique, OcorrenciaDeInssSobreSalariosDevidos>();
        OcorrenciaInssUnique key = null;
        for (OcorrenciaDeInssSobreSalariosDevidos ocorrencia : collection) {
            key = new OcorrenciaInssUnique(ocorrencia);
            this.map.put(key, ocorrencia);
        }
        return this;
    }

    @Override
    public OcorrenciaDeInssSobreSalariosDevidos search(OcorrenciaInssUnique key) {
        return this.map.get(key);
    }
}

File: src/lib/pjecalc/core/dominio/calculo/inss/sobresalarios/ocorrencia-de-inss-salario-devido-optimizer-list-search-unique.ts
/**
 * PJe-Calc v2.15.1 — OcorrenciaDeInssSalarioDevidoOptimizerListSearchUnique
 * Porte 1:1 de: br.jus.trt8.pjecalc.negocio.dominio.calculo.inss.sobresalarios.OcorrenciaDeInssSalarioDevidoOptimizerListSearchUnique
 *
 * Ref Java: pjecalc-fonte/.../sobresalarios/OcorrenciaDeInssSalarioDevidoOptimizerListSearchUnique.java
 *
 * Indexa OcorrenciaDeInssSobreSalariosDevidos por (competencia + decimoTerceiro).
 */
import { OptimizerListSearchUnique } from '../../../../comum/optimizer-list-search-unique';
import { OcorrenciaInssUnique } from './ocorrencia-inss-unique';
import type { OcorrenciaDeInssSobreSalariosDevidos } from './ocorrencia-de-inss-sobre-salarios-devidos';

export class OcorrenciaDeInssSalarioDevidoOptimizerListSearchUnique extends OptimizerListSearchUnique<OcorrenciaInssUnique, OcorrenciaDeInssSobreSalariosDevidos> {
  private map: Map<string, OcorrenciaDeInssSobreSalariosDevidos> = new Map();

  init(collection: Iterable<OcorrenciaDeInssSobreSalariosDevidos>): this {
    this.map = new Map();
    for (const ocorrencia of collection) {
      const key = new OcorrenciaInssUnique(ocorrencia);
      this.map.set(key.toKey(), ocorrencia);
    }
    return this;
  }

  search(key: OcorrenciaInssUnique): OcorrenciaDeInssSobreSalariosDevidos | null {
    return this.map.get(key.toKey()) ?? null;
  }
}
